src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,451 @@
+			return;
+		}
+		// Restart spin sample
+		spin.pos = 0;
+	}
+}
+
+void DiskNoiseDevice::PlaySeek()
+{
+	if (!enable_disk_noise) {
+		return;
+	}
+
+	const size_t index = ChooseWeightedSeekIndex();
+	if (index >= seek.samples.size() ||
+	    seek.samples[index].empty()) {
+		return;
+	}
+
+	// Check if the sample is still playing and don't interrupt if it does
+	if (!seek.current_sample.empty() &&
+	    seek.pos + 1 < seek.current_sample.size()) {
+		return;
+	}
+
+	seek.current_sample = seek.samples[index];
+	seek.pos            = 0;
+}
+
+void DiskNoiseDevice::Shutdown()
+{
+	std::lock_guard<std::mutex> lock(disk_noises->device_mutex);
+	disk_noises->active_devices.erase(
+	        std::remove(disk_noises->active_devices.begin(),
+	                    disk_noises->active_devices.end(),
+	                    this),
+	        disk_noises->active_devices.end());
+}
+
+static void disknoise_init(Section* section)
+{
+	disk_noises = std::make_unique<DiskNoises>();
+
+	assert(section);
+	const auto prop = static_cast<Section_prop*>(section);
+
+	const bool enable_floppy_disk_noise = prop->Get_bool("floppy_disk_noise");
+	const bool enable_hard_disk_noise = prop->Get_bool("hard_disk_noise");
+
+	const auto spin_up = "hdd_spinup.flac";
+	const auto spin    = "hdd_spin.flac";
+	std::vector<std::string> hdd_seek_samples;
+	for (int i = 1; i <= 9; ++i) {
+		hdd_seek_samples.push_back("hdd_seek" + std::to_string(i) + ".flac");
+	}
+
+	const auto floppy_spin_up = "fdd_spinup.flac";
+	const auto floppy_spin    = "fdd_spin.flac";
+	std::vector<std::string> floppy_seek_samples;
+	for (int i = 1; i <= 9; ++i) {
+		floppy_seek_samples.push_back("fdd_seek" + std::to_string(i) + ".flac");
+	}
+	disk_noises->Initialize(enable_floppy_disk_noise,
+	                        enable_hard_disk_noise,
+	                        spin_up,
+	                        spin,
+	                        hdd_seek_samples,
+	                        floppy_spin_up,
+	                        floppy_spin,
+	                        floppy_seek_samples);
+}
+
+static void disknoise_shutdown([[maybe_unused]] Section* section)
+{
+	disk_noises->Shutdown();
+}
+
+void init_disknoise_dosbox_settings(Section_prop& secprop)
+{
+	constexpr auto only_at_start = Property::Changeable::OnlyAtStart;
+
+	auto* bool_prop = secprop.Add_bool("hard_disk_noise", only_at_start, false);
+	bool_prop->Set_help(
+	        "Enables emulated noises for hard disks.\n"
+	        "This plays disk spinning and seek noises using pre-recorded samples.\n"
+	        "This can be combined with the 'hdd_io_speed' setting for an'\n"
+	        "authentic experience. This feature is disabled by default.\n");
+
+	bool_prop = secprop.Add_bool("floppy_disk_noise", only_at_start, false);
+	bool_prop->Set_help(
+	        "Enables emulated noises for floppy disks.\n"
+	        "This can be combined with the 'fdd_io_speed' setting for an'\n"
+	        "authentic experience. This feature is disabled by default.\n");
```

Comment:
Nice descriptions 👍🏻 Let's make a few minor adjustments and then they're good to go:

```cpp
"Enable emulated hard disk sounds ('off' by default).\n"
"Plays spinning disk and seek noise sounds when enabled. It's recommended to\n"
"set 'hard_disk_speed' to lower than 'maximum' for an authentic experience."
```

```cpp
"Enable emulated floppy disk sounds ('off' by default).\n"
"Plays spinning disk and seek noise sounds when enabled. It's recommended to\n"
"set 'floppy_disk_speed' to lower than 'maximum' for an authentic experience."
```